plugins/CoreHome/Controller.php

```diff
@@ -185,10 +187,44 @@ protected function setDateTodayIfWebsiteCreatedToday()
     public function index()
     {
         $this->setDateTodayIfWebsiteCreatedToday();
+
+        if ($this->noRecordedData()) {
+            return $this->showSiteNoData();
+        }
+
         $view = $this->getDefaultIndexView();
         return $view->render();
     }
 
+    private function noRecordedData()
+    {
+        $sql = "SELECT COUNT(*)
+                FROM " . Common::prefixTable('log_visit') . "
+                WHERE idsite=?";
+
+        $count = Db::fetchOne($sql, array($this->idSite));
+
+        return $count == 0;
+    }
```

Comment:
Thanks! I've done the changes and moved everything to SitesManager and used the [`Request.dispatch` event](https://github.com/piwik/piwik/blob/no-data-welcome/core/FrontController.php#L577-587) as it seems to be exactly made for that. `Controller.CoreHome.index` didn't allow to change the controller.

> Maybe also apply a Limit 1 to the query as we only want to know whether there was at least one tracking request.

the limit would apply to the rows returned, given COUNT(*) returns one row it has no effect.